### PR TITLE
Answer:1

### PR DESCRIPTION
**Pull request by Gw0zdziu**

### CODE REVIEW

File: apps/angular/projection/src/app/ui/card/card.component.ts [hunk 1]
```diff
@@ -12,21 +15,13 @@ import { ListItemComponent } from '../list-item/list-item.component';
...
-        width="200px" />
-
+      <ng-content select="[image]"></ng-content>
       <section>
-        <app-list-item
-          *ngFor="let item of list"
-          [name]="item.firstName"
-          [id]="item.id"
-          [type]="type"></app-list-item>
+        @for (item of list; track item.id) {
```
Comment (tomalaforge) [inline apps/angular/projection/src/app/ui/card/card.component.ts:20]: 👍

File: apps/angular/projection/src/app/ui/card/card.component.ts [hunk 2]
```diff
@@ -37,25 +32,18 @@ import { ListItemComponent } from '../list-item/list-item.component';
     </div>
   `,
   standalone: true,
-  imports: [NgIf, NgFor, ListItemComponent],
+  imports: [ListItemComponent, NgTemplateOutlet],
 })
 export class CardComponent {
```
Comment (tomalaforge) [inline apps/angular/projection/src/app/ui/card/card.component.ts:37]: use a generic, it's nicer than using any

File: apps/angular/projection/src/app/ui/card/card.component.ts [hunk 3]
```diff
@@ -37,25 +32,18 @@ import { ListItemComponent } from '../list-item/list-item.component';
...
   @Input() list: any[] | null = null;
-  @Input() type!: CardType;
   @Input() customClass = '';
+  @Output() AddNewRecordEmitter = new EventEmitter<null>();
+  @Input() ListItemTemplate!: TemplateRef<any>;
```
Comment (tomalaforge) [inline apps/angular/projection/src/app/ui/card/card.component.ts:41]: you can use contentchild to project template into your child

File: apps/angular/projection/src/app/ui/card/card.component.ts [hunk 4]
```diff
@@ -37,25 +32,18 @@ import { ListItemComponent } from '../list-item/list-item.component';
...
 
   CardType = CardType;
 
-  constructor(
-    private teacherStore: TeacherStore,
-    private studentStore: StudentStore,
-  ) {}
-
+  constructor() {}
```
Comment (tomalaforge) [inline apps/angular/projection/src/app/ui/card/card.component.ts:45]: delete all useless element

File: apps/angular/projection/src/app/ui/list-item/list-item.component.ts [hunk 5]
```diff
@@ -1,35 +1,24 @@
...
+import { Component, EventEmitter, Input, Output } from '@angular/core';
 
 @Component({
   selector: 'app-list-item',
   template: `
     <div class="border-grey-300 flex justify-between border px-2 py-1">
-      {{ name }}
-      <button (click)="delete(id)">
+      <ng-content></ng-content>
+      <button (click)="delete(item.id)">
```
Comment (tomalaforge) [inline apps/angular/projection/src/app/ui/list-item/list-item.component.ts:8]: you don't need to pass the item. delete can be a simple event emitter.

File: apps/angular/projection/src/app/component/city-card/city-card.component.ts [hunk 6]
```diff
@@ -1,13 +1,50 @@
...
+  constructor(
+    private http: FakeHttpService,
+    private store: CityStore,
+  ) {}
 
-  ngOnInit(): void {}
+  ngOnInit(): void {
+    this.http.fetchCities$.subscribe((s) => this.store.addAll(s));
+
+    this.store.cities$.subscribe((s) => this.cities.set(s));
```
Comment (tomalaforge) [inline apps/angular/projection/src/app/component/city-card/city-card.component.ts:40]: no prefer declarative over imperative.

you can transform your state to be signal or use toSignal l.31

File: apps/angular/projection/src/app/ui/card/card.component.ts (hunk 3 above)
Comment (Gw0zdziu) [reply apps/angular/projection/src/app/ui/card/card.component.ts:41]: done

File: apps/angular/projection/src/app/ui/card/card.component.ts (hunk 4 above)
Comment (Gw0zdziu) [reply apps/angular/projection/src/app/ui/card/card.component.ts:45]: done

File: apps/angular/projection/src/app/ui/card/card.component.ts (hunk 2 above)
Comment (Gw0zdziu) [reply apps/angular/projection/src/app/ui/card/card.component.ts:37]: done

File: apps/angular/projection/src/app/ui/list-item/list-item.component.ts (hunk 5 above)
Comment (Gw0zdziu) [reply apps/angular/projection/src/app/ui/list-item/list-item.component.ts:8]: done

File: apps/angular/projection/src/app/component/city-card/city-card.component.ts (hunk 6 above)
Comment (Gw0zdziu) [reply apps/angular/projection/src/app/component/city-card/city-card.component.ts:40]: done

File: apps/angular/projection/src/app/ui/list-item/list-item.component.ts (hunk 5 above)
Comment (Gw0zdziu) [reply apps/angular/projection/src/app/ui/list-item/list-item.component.ts:8]: done

**Pull request closed without merging**